File: app/src/main/java/com/wateria/Database/PlantEntity.java
package com.wateria.Database;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class PlantEntity {

    // IMPORTANT: non primitive types are @nullable.
    // int     = notNull
    // INTEGER = nullable

    @PrimaryKey
    public int id;

    public String name;

    public Integer wat_freq;

    public Integer light;

    public String url;

}

File: data/src/main/kotlin/com/wateria/data/database/entity/PlantEntity.kt
package com.wateria.data.database.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "plants")
data class PlantEntity(
    @PrimaryKey val id: String,
    val name: String,
    @ColumnInfo(name = "icon_key") val iconKey: String,
    @ColumnInfo(name = "watering_interval_days") val wateringIntervalDays: Int,
    @ColumnInfo(name = "next_watering_epoch_day") val nextWateringEpochDay: Long,
    @ColumnInfo(name = "created_at_epoch_millis") val createdAtEpochMillis: Long,
    @ColumnInfo(name = "updated_at_epoch_millis") val updatedAtEpochMillis: Long
)
